hs_core/views/resource_rest_api.py

```diff
@@ -0,0 +1,622 @@
+
+    :rtype:  json string
+
+    :return:  a paginated list of resources with data for resource id, title, resource type, creator, sharing status,
+    date created, date last updated, resource bag url path, and science metadata url path
+
+    example return json format:
+
+        {   "count":n
+            "next": link to next page
+            "previous": link to previous page
+            "results":[
+                    {"resource_type": resource type, "resource_title": resource title, "resource_id": resource id,
+                    'creator': creator name, 'sharing_status': Public or Private, 'date_created': date resource created,
+                    'date_last_update': date resource last updated, 'bag': link to bag file, 'science_metadata':
+                    link to science metadata},
+                    {"resource_type": resource type, "resource_title": resource title, "resource_id": resource id,
+                    'creator': creator name, 'sharing_status': Public or Private, 'date_created': date resource created,
+                    'date_last_update': date resource last updated, 'bag': link to bag file, 'science_metadata':
+                    link to science metadata},
+            ]
+        }
+
+    :raises:
+
+    &nbsp;&nbsp;&nbsp;ValidationError: return json format: {'parameter-1':['error message-1'], 'parameter-2': ['error message-2'], .. }
+
+    &nbsp;&nbsp;&nbsp;HTTP status code:400
+    """
+
+    pagination_class = PageNumberPagination
+
+    def get_queryset(self):
+        return self._get_resource_list()
+
+    def get_serializer_class(self):
+        return serializers.ResourceListItemSerializer
+
+    # This class uses the default list() method from the ListAPIView class to get the paginated results
+    # The list() method is executed automatically on a GET REST call
+
+    def _get_resource_list(self):
+        resource_list_request_validator = serializers.ResourceListRequestValidator(data=self.request.query_params)
+        if not resource_list_request_validator.is_valid():
+            raise ValidationError(detail=resource_list_request_validator.errors)
+
+        filter_parms = resource_list_request_validator.validated_data
+        filter_parms['user'] = (self.request.user if self.request.user.is_authenticated() else None)
+        if len(filter_parms['types']) == 0:
+            filter_parms['types'] = None
+
+        filter_parms['public'] = not self.request.user.is_authenticated()
+
+        filter_parms['keywords'] = None
+
+        filtered_res_list = []
+        filter_parms = dict(filter_parms)
+
+        resource_table = hydroshare.get_resource_list(**filter_parms)
+        res = set()
+
+        for resources in resource_table.values():
+            res = res.union(resources)
+
+        res_list = list(res)
+        for res in res_list:
+            try:
+                utils.authorize(self.request, res.short_id, view=True)
+            except:
+                pass
```

Comment:
why are you ignoring errors here?

Also, you should never use a bare `except:` as it can catch things like MemoryError and KeyboardException, and cause a process to fail to terminate properly.